src/sap/fhir/model/r4/FHIRListBinding.js

```diff
@@ -391,28 +398,36 @@ sap.ui.define([
 	FHIRListBinding.prototype.checkUpdate = function(bForceUpdate, mChangedEntities, sMethod) {
 		var oBindingInfo = this.oModel.getBindingInfo(this.sPath, this.oContext, this.bUnique);
 		var mResources = oBindingInfo && mChangedEntities && mChangedEntities[oBindingInfo.getResourceType()];
-		if (mResources && sMethod && sMethod !== HTTPMethod.GET){
-			for (var sId in mResources){
-				if (!this.isRelative()){
-					if (sMethod === HTTPMethod.DELETE){
-						this.iTotalLength--;
-						this.aKeysServerState.splice(this.aKeysServerState.indexOf(oBindingInfo.getResourceType() + "/" + sId), 1);
-					} else if (sMethod === HTTPMethod.POST){
+		if (mResources && sMethod && sMethod !== HTTPMethod.GET) {
+			var sResType = oBindingInfo.getResourceType();
+			for (var sId in mResources) {
+				var sResPath = sResType + "/" + sId;
+				if (!this.isRelative()) {
+					if (sMethod === HTTPMethod.DELETE) {
+						// check for context
+						if (this.oModel.mRemovedResources[sResType] && this.oModel.mRemovedResources[sResType].indexOf(sResPath) > -1) {
```

Comment:
for the `mChangedResources` we use the local function parameter instead of the value stored in the model. Whats your opinion on adding a new parameter to the function `mRemovedEntities`

Reply:
as discussed this is not necessary